lib/matplotlib/colors.py

```diff
@@ -1992,3 +1994,49 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     norm = BoundaryNorm(levels, ncolors=n_data_colors)
     return cmap, norm
+
+
+def brighten_color(color, frac):
+    """A color helper utility to either darken or brighten the given color.
+
+    This function first converts the given color to RGB using
+    `~.colorConverter` and then to
+    HSL using `colorsys.rgb_to_hsl`.  The lightness is modified according
+    to the given fraction, clipped to be between 0 and 1, and then converted
+    back to RGB using `colorsys.hsl_to_rgb`:
+    L = np.clip(L0 * (1.0 + frac), 0., 1.), where L0 is the original
```

Comment:
I'm not convinced that this operation really makes a lot of sense, e.g. is going from L=0.1 to L=0.2 (frac=1) really "equivalent" from going to L=0.5 to L=1?  If anything I think this should something like a probit or logit scale.

Reply:
See [this comment](https://github.com/matplotlib/matplotlib/pull/9985#discussion_r156460588)

Reply:
I don’t feel strongly about this.  Do you want black to stay black or go grey?  I could go either way. I still prefer fractional change because I think that’s what most people would mean when they say to lighten an image.  I’d you want the histogram to move right but black would still stay black. 

We could also just forget any algorithm and just ask the user for the L value they want perhaps w an accompanying way to get L so they know where they are starting from.

Reply:
I would indeed prefer a design where we expose a color class with a method such as `color.hsl_replace(l=...)`, `color.hsv_replace(s=...)` (and allowing multiple kwargs of course, just leaving the ones not specified untouched).